### PR TITLE
[CALCITE-4474] SqlSimpleParser inner Tokenizer should not recognize the sql of TokenType.ID or some keywords in some case

### PR DESCRIPTION
**Pull request by wangjie-fourth**
Hi, this PR is uesd for solve [CALCITE-4474](https://issues.apache.org/jira/browse/CALCITE-4474)

If you think something is wrong, please comment below, i will keep eye on it.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/sql/advise/SqlSimpleParserTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * A <code>SqlSimpleParserTest</code> is a unit-test for
+ * {@link org.apache.calcite.sql.advise.SqlSimpleParser}.
+ */
+public class SqlSimpleParserTest {
```
Comment (amaliujia) [inline core/src/test/java/org/apache/calcite/sql/advise/SqlSimpleParserTest.java:31]: Per the suggestion in JIRA, can you move this test to SqlAdvisorTest

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java [hunk 2]
```diff
@@ -1585,4 +1587,17 @@ private void testSimpleParserQuotedIdImpl() {
...
   }
+
+  private static SqlSimpleParser simpleParser =
+      new SqlSimpleParser("_suggest_", SqlParser.Config.DEFAULT);
+
+  @Test public void testFilterComment() {
+    // when SqlSimpleParser.Tokenizer#nextToken() method parse sql,
+    // ignore the  "--" after 10.0, this is a comment,
+    // but Tokenizer#nextToken() does not recognize it
+    String sqlWithComment = "select count(*) from a where price > 10.0-- this is comment n"
```
Comment (amaliujia) [inline core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java:1598]: The last `n` should be `\n`?

**Comment by amaliujia** [comment]
Overall LGTM. Left a comment on the test case.

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java [hunk 3]
```diff
@@ -1585,4 +1587,17 @@ private void testSimpleParserQuotedIdImpl() {
     assertSimplify(sql, simplified);
     assertComplete(sql, EXPR_KEYWORDS, Collections.singletonList("TABLE(a)"), DEPT_COLUMNS);
   }
+
+  private static SqlSimpleParser simpleParser =
+      new SqlSimpleParser("_suggest_", SqlParser.Config.DEFAULT);
```
Comment (julianhyde) [inline core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java:1592]: Make this a local variable, not a static field. Static fields use memory after the test has finished, and may not be reentrant.

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java [hunk 4]
```diff
@@ -1585,4 +1587,47 @@ private void testSimpleParserQuotedIdImpl() {
...
+    // ignore the  "--" after 10.0, this is a comment,
+    // but Tokenizer#nextToken() does not recognize it
+    originSql = baseOriginSql + "where price > 10.0-- this is comment \n"
+          + " -- comment ";
+    actualSql = simpleParser.simplifySql(originSql);
+    assertThat(actualSql, equalTo(baseResultSql + "WHERE price > 10.0"));
+
+    originSql = baseOriginSql + "where column_b='/* this is not comment */'";
+    actualSql = simpleParser.simplifySql(originSql);
+    assertThat(actualSql, equalTo(baseResultSql + "WHERE column_b= '/* this is not comment */'"));
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java:1623]: Please factor this out to a helper method that would take `originSql`, and `expectedSql`.
Then the method should add `originSql`into the assertion message.

**Comment by wangjie-fourth** [comment]
@amaliujia @julianhyde @vlsi

**Comment by wangjie-fourth** [comment]
@amaliujia @julianhyde @vlsi Hi,thanks for your review, I have done what your said.Is there anything else I need to do?

**Comment by amaliujia** [comment]
@wangjie-fourth 

Thanks. I will give another pass soon! If there is not major comment I will merge this PR (for minors I can update directly).

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java [hunk 5]
```diff
@@ -1585,4 +1587,47 @@ private void testSimpleParserQuotedIdImpl() {
...
+    filterCommentHelper(originSql, baseResultSql + "WHERE column_b= '/* this is not comment */'");
+
+    originSql = baseOriginSql + "where column_b='2021 --this is not comment'";
+    filterCommentHelper(originSql, baseResultSql + "WHERE column_b= '2021 --this is not comment'");
+
+    originSql = baseOriginSql + "where column_b='2021--this is not comment'";
+    filterCommentHelper(originSql, baseResultSql + "WHERE column_b= '2021--this is not comment'");
+  }
+
+  private void filterCommentHelper(String originSql, String expectedSql) {
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java:1626]: Please rename the method so it is clear what it does.
In most cases, `helper` in the method name does not add much value.

For instance: `assertSimplifySql`

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java [hunk 6]
```diff
@@ -1585,4 +1587,47 @@ private void testSimpleParserQuotedIdImpl() {
...
+    SqlSimpleParser simpleParser =
+        new SqlSimpleParser("_suggest_", SqlParser.Config.DEFAULT);
+
+    String actualSql = simpleParser.simplifySql(originSql);
+    assertThat(originSql, actualSql, equalTo(expectedSql));
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java:1631]: Please add clarification to the message, for instance: `"simpleParser.simplifySql(" + originSql + ")"`

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java (hunk 5 above)
Comment (wangjie-fourth) [reply core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java]: thanks, it is done

File: core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java (hunk 6 above)
Comment (wangjie-fourth) [reply core/src/test/java/org/apache/calcite/sql/test/SqlAdvisorTest.java]: thanks, it is done

**Comment by amaliujia** [comment]
Thanks for all the code review so far! If there is no more comments with a day, I will merge this PR.

**Comment by amaliujia** [comment]
Fixed via https://github.com/apache/calcite/commit/a16a47351aa25caf538c1955edd171ec871569f7

**Pull request closed without merging**